### PR TITLE
Add more data generators. Closes #6.

### LINKED ISSUE #6: Add data generators for S3, EBS, and data transfers
Generate data  for storage with S3, EBS
Generate "other" data with data transfers.

### PR DESCRIPTION
**Pull request by chambridge**
* S3 generator for storage data
* EBS generator for storage data
* Data Transfer generator for other data

### CODE REVIEW

File: nise/generators/ebs_generator.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        row['lineItem/Operation'] = 'CreateVolume'
+        row['lineItem/ResourceId'] = 'vol-{}'.format(self.fake.ean8())  # pylint: disable=no-member
+        row['lineItem/UsageAmount'] = str(amount)
+        row['lineItem/CurrencyCode'] = 'USD'
+        row['lineItem/UnblendedRate'] = str(rate)
+        row['lineItem/UnblendedCost'] = str(cost)
+        row['lineItem/BlendedRate'] = str(rate)
+        row['lineItem/BlendedCost'] = str(cost)
+        row['lineItem/LineItemDescription'] = description
+        row['product/ProductName'] = 'Amazon Elastic Compute Cloud'
```
Comment (blentz) [inline nise/generators/ebs_generator.py]: Wrong product name for EBS.

File: nise/generators/ebs_generator.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        rate = round(uniform(0.02, 0.16), 3)
+        amount = uniform(0.2, 300.99)
+        cost = amount * rate
+        location, _, storage_region = self._get_location()
+        description = '${} per GB-Month of snapshot data stored - {}'.format(rate, location)
+        burst, max_iops, max_thru, max_vol_size, vol_backed, vol_type = \
+            self._get_storage()
+
+        row['lineItem/ProductCode'] = 'AmazonEC2'
```
Comment (blentz) [inline nise/generators/ebs_generator.py]: Wrong product code for EBS.

File: nise/generators/s3_generator.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        row = self._add_common_usage_info(row, start)
+
+        rate = round(uniform(0.02, 0.06), 3)
+        amount = uniform(0.2, 6000.99)
+        cost = amount * rate
+        location, avail_zone, storage_region = self._get_location()
+        description = '${} per GB-Month of snapshot data stored - {}'.format(rate, location)
+        amazon_resource_name = self._get_arn(avail_zone)
+
+        row['lineItem/ProductCode'] = 'AmazonEC2'
```
Comment (blentz) [inline nise/generators/s3_generator.py]: Wrong product code for S3

File: nise/generators/s3_generator.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        row['lineItem/Operation'] = 'CreateSnapshot'
+        row['lineItem/ResourceId'] = amazon_resource_name
+        row['lineItem/UsageAmount'] = str(amount)
+        row['lineItem/CurrencyCode'] = 'USD'
+        row['lineItem/UnblendedRate'] = str(rate)
+        row['lineItem/UnblendedCost'] = str(cost)
+        row['lineItem/BlendedRate'] = str(rate)
+        row['lineItem/BlendedCost'] = str(cost)
+        row['lineItem/LineItemDescription'] = description
+        row['product/ProductName'] = 'Amazon Elastic Compute Cloud'
```
Comment (blentz) [inline nise/generators/s3_generator.py]: Wrong product name for S3

File: nise/generators/s3_generator.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        row['product/location'] = location
+        row['product/locationType'] = 'AWS Region'
```
Comment (blentz) [inline nise/generators/s3_generator.py]: IIRC, S3 objects have no region. There are regional endpoints you can use to access S3, but I don't think S3 objects are tied to a specific region (except GovCloud).

File: nise/generators/s3_generator.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        row['product/productFamily'] = 'Storage Snapshot'
+        row['product/servicecode'] = 'AmazonEC2'
```
Comment (blentz) [inline nise/generators/s3_generator.py]: Wrong service code for S3

File: nise/generators/s3_generator.py (hunk 3 above)
Comment (chambridge) [reply nise/generators/s3_generator.py]: I took this from the cloud health data posted.

File: nise/generators/s3_generator.py (hunk 4 above)
Comment (chambridge) [reply nise/generators/s3_generator.py]: I took this from the cloud health data posted.

File: nise/generators/s3_generator.py (hunk 5 above)
Comment (chambridge) [reply nise/generators/s3_generator.py]: I took this from the cloud health data posted.

File: nise/generators/s3_generator.py (hunk 6 above)
Comment (chambridge) [reply nise/generators/s3_generator.py]: I took this from the cloud health data posted.

**Comment by chambridge** [comment]
I like your Generator metadata factory idea in theory, but I tend to think its overly heavy at this point.

Once this PR is in, I'm happy for the short term on basic mock data generation as we cover basic data for what we need to test the dashboard reports. I'm not looking to produce mock data for a huge variety of products.

Could you point to the code duplication that you don't like as I've pulled most of it up into the Abstract generator class. Just happens that services I focused on have many similar columns.

**Comment by chambridge** [comment]
Opened the following issue to capture the Generator Factory idea:
https://github.com/project-koku/nise/issues/8

**Pull request merged**